### PR TITLE
Feature/#341 update access keys

### PR DESCRIPTION
**Pull request by oconn**
# PR for issue #341 #.

## Ready for a PR?
 - [x] I have updated the CHANGELOG with an entry including a description, a link to my profile, and a link to the issue ticket. This change is filed under an Enhancement or Bug Fix, is shown under the relevant release tag name, and is included in my PR branch. 
 
## Implementation Notes:

- Added handler to support updating access keys
- Added error responses to login (The error is located in a query parameter in place of the access token `login-error`)

## Breaking changes:

- None

### CODE REVIEW

File: src/wombats/handlers/access_key.clj [hunk 1]
```diff
@@ -19,12 +20,12 @@
...
    :required true})
 
-(def ^:private add-access-key-body-params
+(def ^:private access-key-body-params
   {:name "access-key-body"
    :in "body"
    :description "values for an access key"
    :required true
-   :default (str add-access-key-body-sample)
+   :default (str access-key-body-sample)
```
Comment (emilyseibert) [inline src/wombats/handlers/access_key.clj]: do you need a str here?

File: src/wombats/handlers/access_key.clj [hunk 2]
```diff
@@ -68,13 +70,14 @@
...
            :responses {:200 {:description "add-access-key response"}}}}})
 
-(defn- format-new-access-key-request
+(defn- format-access-key-request
   [access-key]
-  (-> access-key
-      (update :access-key/expiration-date #(when % (read-string (str "#inst \"" % "\""))))))
+  (when access-key
+    (-> access-key
+        (update :access-key/expiration-date #(when % (read-string (str "#inst \"" % "\"")))))))
```
Comment (emilyseibert) [inline src/wombats/handlers/access_key.clj]: line too long

File: src/wombats/handlers/access_key.clj [hunk 3]
```diff
@@ -143,15 +145,39 @@
...
+  [{user-ref :db/id}
+   current-access-key
+   updated-access-key]
+
+  (let [{:keys [:access-key/number-of-uses
+                :access-key/max-number-of-uses] :as access-key}
+        (merge current-access-key updated-access-key)]
+
+    (when (> number-of-uses max-number-of-uses)
+      (wombat-error {:code :handlers.access_key.update-access-key-fields/max-number-of-keys
```
Comment (emilyseibert) [inline src/wombats/handlers/access_key.clj]: too long

File: src/wombats/handlers/access_key.clj [hunk 4]
```diff
@@ -143,15 +145,39 @@
...
+     (let [update-access-key (dao/get-fn :update-access-key context)
+           access-key-id (get-in request [:path-params :access-key-id])
+           current-access-key ((dao/get-fn :get-access-key-by-id context) access-key-id)
+           new-access-key (->> (:edn-params request)
+                               (format-access-key-request)
+                               (sutils/validate-input ::new-access-key)
+                               (update-access-key-fields (get-current-user context)
+                                                         current-access-key))]
+
        (assoc context :response (assoc response
```
Comment (emilyseibert) [inline src/wombats/handlers/access_key.clj]: too long

File: src/wombats/handlers/access_key.clj (hunk 1 above)
Comment (oconn) [inline src/wombats/handlers/access_key.clj]: Yep, it converts the map to a string to be used in swagger.

File: src/wombats/handlers/auth.clj [hunk 5]
```diff
@@ -149,9 +150,11 @@
                  ;; Check if the user has a vaild access key
                  (if (:user/access-key updated-user)
                    (redirect-home context referer user-access-token)
-                   (failed-redirect))))
-             (failed-redirect)))
-         (failed-redirect))))))
+                   (failed-redirect "Wombats is in Alpha. You'll need to request an early access token to get started now."))))
```
Comment (emilyseibert) [inline src/wombats/handlers/auth.clj]: too long. also awk

Comment (emilyseibert) [inline src/wombats/handlers/auth.clj]: "Oops! What's the magic word? Please request an access key from an administrator."

File: src/wombats/handlers/auth.clj [hunk 6]
```diff
@@ -149,9 +150,11 @@
...
+             (failed-redirect "We were unable to lookup your Github Account.")))
+         (do
+           (log/error (str "Signin request has been tampored with.\n\n" request))
```
Comment (emilyseibert) [inline src/wombats/handlers/auth.clj]: sign in

**Pull request merged**